### PR TITLE
Support nullable boolean column in schema enforcement

### PR DESCRIPTION
**Pull request by serena-ruan**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/serena-ruan/mlflow/pull/10727?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10727/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10727
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?
Allow boolean column with nulls pass schema enforcement 

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/pyfunc/test_pyfunc_schema_enforcement.py [hunk 1]
```diff
@@ -1611,6 +1611,31 @@ def test_enforce_schema_in_python_model_predict(sample_params_basic, param_schem
...
 
+def test_schema_enforcement_all_feature_types_pandas():
+    data = {
+        "f1": list(range(0, 200)),
+        "f2": [(i % 3) == 0 for i in range(0, 200)],
+        "f3": [f"string id: {i}" for i in range(0, 200)],
+        "f4": [pd.Timestamp(f"2020-07-14 00:{i//60}:{i%60}") for i in range(1, 201)],
+        "f1_nulls": [True if i % 4 != 0 else None for i in range(200)],
+        "f2_nulls": ["test" if i % 4 != 1 else None for i in range(200)],
+        "f3_nulls": [i if i % 4 != 2 else None for i in range(200)],
```
Comment (harupy) [inline tests/pyfunc/test_pyfunc_schema_enforcement.py:1622]: Do we need 200 samples? can we hardcode values?

Comment (serena-ruan) [reply tests/pyfunc/test_pyfunc_schema_enforcement.py:1622]: Lol yes, I'll modify. I just copied AutoML's example 😆

**Review by harupy — APPROVED** [review]
LGTM!

File: tests/pyfunc/test_pyfunc_schema_enforcement.py [hunk 2]
```diff
@@ -1611,6 +1611,31 @@ def test_enforce_schema_in_python_model_predict(sample_params_basic, param_schem
     ] == np.datetime64("2023-06-26 00:00:00")
 
 
+def test_schema_enforcement_all_feature_types_pandas():
+    data = {
+        "f1": [1, 2, 3],
+        "f2": [True, False, False],
```
Comment (harupy) [inline tests/pyfunc/test_pyfunc_schema_enforcement.py:1617]: can we rename columns? for example, f2 -> bool

**Pull request merged**